lib/model.js

```diff
@@ -1253,15 +1269,57 @@ Model.prototype.findAll = function(options) {
       return this.runHooks('beforeFindAfterOptions', options);
     }
   }).then(function() {
+    originalOptions = optClone(options);
     options.tableNames = Object.keys(tableNames);
+
     return this.QueryInterface.select(this, this.getTableName(options), options);
   }).tap(function(results) {
     if (options.hooks) {
       return this.runHooks('afterFind', results, options);
     }
+  }).then(function (results) {
+    return Model.$findSeperate(results, originalOptions);
   });
 };
 
+Model.$findSeperate = function(results, options) {
+  if (!options.include || options.raw || !results) return Promise.resolve(results);
+
+  var original = results;
+  if (options.plain) results = [results];
+
+  return Promise.map(options.include, function (include) {
+    if (!include.seperate) {
+      return Model.$findSeperate(
+        results.reduce(function (memo, result) {
+          var associations = result.get(include.association.as);
+          if (!Array.isArray(associations)) associations = [associations];
+
+          return memo.concat(associations);
+        }, []),
+        _.assign(
```

Comment:
The creating of options? The last argument they take is seperate in each. The omit could however be changed to only once though.